util/semaphore-group/semaphore_group.go

```diff
@@ -0,0 +1,39 @@
+package semaphoregroup
```

Comment:
Better place inside the peer package.

Reply:
I am moving this package from another project and the util seems the most appropriated